react-native/react/actions/login2.js

```diff
@@ -299,3 +306,123 @@ export function setDeviceName (name) {
     }, 1000)
   }
 }
+*/
+
+// Show a user/pass screen, call cb(user, passphrase) when done
+// title/subTitle to customize the screen
+export const askForUserPass = (title, subTitle, cb, hidePass = false) => {
+  return (dispatch) => {
+    dispatch({
+      type: Constants.actionAskUserPass,
+      title,
+      subTitle,
+      hidePass,
+      onSubmit: (username, passphrase) => {
+        dispatch({
+          type: Constants.actionSetUserPass,
+          username,
+          passphrase
+        })
+
+        cb()
+      }
+    })
+
+    dispatch(routeAppend({
```

Comment:
it will be impossible to get hot reloading to work perfectly in all states (especially here). We have a back and forth with the rpc which has session ticks which you can't time travel (since we can't time travel the go side). it would be nice to time travel anywhere but its not something i want to make work and make the code more complex